sw/device/lib/drivers/gpio/gpio.h

```diff
@@ -1,282 +1,369 @@
- * setting and includes the effects of the input noise filter and the load on
- * the pin.
- *
- * @param gpio A GPIO handle.
- * @param pin A GPIO pin.
- * @param[out] state Pin value.
- * @return The result of the operation.
+ * @brief enable falling edge interrupt for the given pin.
+ * @param gpio_pin_number_t specify pin number
  */
-gpio_result_t gpio_read(const gpio_t *gpio, gpio_pin_t pin,
-                                bool *state);
+gpio_result_t gpio_intr_en_fall (gpio_pin_number_t pin);
 
 /**
- * Writes to a pin.
- *
- * The actual value on the pin depends on the output enable setting.
- *
- * @param gpio A GPIO handle.
- * @param pin A GPIO pin.
- * @param state Value to write.
- * @return The result of the operation.
+ * @brief disable falling edge interrupt for the given pin.
+ * @param gpio_pin_number_t specify pin number
  */
-gpio_result_t gpio_write(const gpio_t *gpio, gpio_pin_t pin,
-                                 bool state);
-/**
- * Sets output modes of all pins.
- *
- * @param gpio A GPIO handle.
- * @param pin A GPIO pin.
- * @param state Output modes of the pins.
- * @return The result of the operation.
- */
-gpio_result_t gpio_input_set_enabled(const gpio_t *gpio,
-                                                  gpio_pin_t pin,
-                                                  gpio_state_t state);
-/**
- * Enable sampling on GPIO
- * *
- * @param gpio A GPIO handle.
- * @param pin A GPIO pin.
- * @param state Value to write.
- */
-gpio_result_t gpio_input_enabled(const gpio_t *gpio,
-                                          gpio_pin_t pin, gpio_toggle_t state);
-
-
-#ifdef __cplusplus
-}  // extern "C"
-#endif  // __cplusplus
-
-#endif  // GPIO_H_
+gpio_result_t gpio_intr_dis_fall (gpio_pin_number_t pin);
+
+/**
+ * @brief enable logic-high level-sensitive interrupt for the given pin.
+ * @param gpio_pin_number_t specify pin number
+ */
+gpio_result_t gpio_intr_en_lvl_high (gpio_pin_number_t pin);
+
+/**
+ * @brief disable logic-high level-sensitive interrupt for the given pin.
+ * @param gpio_pin_number_t specify pin number
+ */
+gpio_result_t gpio_intr_dis_lvl_high (gpio_pin_number_t pin);
+
+/**
+ * @brief enable logic-low level-sensitive interrupt for the given pin.
+ * @param gpio_pin_number_t specify pin number
+ */
+gpio_result_t gpio_intr_en_lvl_low (gpio_pin_number_t pin);
+
+/**
+ * @brief disable logic-low level-sensitive interrupt for the given pin.
+ * @param gpio_pin_number_t specify pin number
+ */
+gpio_result_t gpio_intr_dis_lvl_low (gpio_pin_number_t pin);
+
+/**
+ * @brief enable the interrupt for the given pin. Type of interrupt 
+ * specified by user.
+ * @param gpio_pin_number_t specify pin number
+ * @param gpio_intr_type_t specify the type of the interrupt
+ */
+gpio_result_t gpio_intr_en (gpio_pin_number_t pin, gpio_intr_type_t type);
+
+// /**
+//  * @brief disable the given type of interrupt on the given pin.
```

Comment:
why do we have different functions that are commented?

Reply:
removed.

Reply:
are they?